src/spdx_tools/spdx3/bump_from_spdx2/relationship.py

```diff
@@ -1,74 +1,277 @@
+        DependencyConditionalityType.OPTIONAL,
+    ),
+    Spdx2_RelationshipType.OTHER: (True, RelationshipType.OTHER),
+    Spdx2_RelationshipType.PACKAGE_OF: (False, RelationshipType.DEPENDS_ON),
+    Spdx2_RelationshipType.PATCH_APPLIED: (True, RelationshipType.PATCH, True),
+    Spdx2_RelationshipType.PATCH_FOR: (True, RelationshipType.PATCH),
+    Spdx2_RelationshipType.PREREQUISITE_FOR: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        None,
+        None,
+        DependencyConditionalityType.PREREQUISITE,
+    ),
+    Spdx2_RelationshipType.PROVIDED_DEPENDENCY_OF: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        LifecycleScopeType.BUILD,
+        None,
+        DependencyConditionalityType.PROVIDED,
+    ),
+    Spdx2_RelationshipType.RUNTIME_DEPENDENCY_OF: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        LifecycleScopeType.RUNTIME,
+        None,
+        None,
+    ),
+    Spdx2_RelationshipType.STATIC_LINK: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        None,
+        SoftwareDependencyLinkType.STATIC,
+        None,
+    ),
+    Spdx2_RelationshipType.TEST_CASE_OF: (True, RelationshipType.TEST_CASE),
+    Spdx2_RelationshipType.TEST_DEPENDENCY_OF: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        LifecycleScopeType.TEST,
+        None,
+        None,
+    ),
+    Spdx2_RelationshipType.TEST_OF: (True, RelationshipType.TEST),
+    Spdx2_RelationshipType.TEST_TOOL_OF: (
+        False,
+        RelationshipType.DEPENDS_ON,
+        LifecycleScopeType.TEST,
+        SoftwareDependencyLinkType.TOOL,
+        None,
+    ),
+    Spdx2_RelationshipType.VARIANT_OF: (True, RelationshipType.VARIANT),
+    Spdx2_RelationshipType.REQUIREMENT_DESCRIPTION_FOR: (True, RelationshipType.REQUIREMENT_FOR),
+    Spdx2_RelationshipType.SPECIFICATION_FOR: (True, RelationshipType.SPECIFICATION_FOR),
+}
 
-def bump_relationship(
-    spdx2_relationship: Spdx2_Relationship,
+
+def bump_relationships(
+    spdx2_relationships: List[Spdx2_Relationship],
     payload: Payload,
     creation_information: CreationInformation,
     document_namespace: str,
-    counter: int,
 ):
-    relationship_type, swap_direction = bump_relationship_type(spdx2_relationship.relationship_type)
+    generated_relationships: Dict[Tuple[str, str], List[Relationship]] = {}
+    for counter, spdx2_relationship in enumerate(spdx2_relationships):
+        relationship = bump_relationship(spdx2_relationship, creation_information, document_namespace, counter)
+        if relationship:
+            generated_relationships.setdefault(
+                (relationship.from_element, relationship.relationship_type.name), []
+            ).append(relationship)
 
-    spdx_id = "#".join([document_namespace, f"SPDXRef-Relationship-{counter}"])
+    for key, relationships in generated_relationships.items():
+        if len(relationships) > 1:
+            merge_relationships_and_add_to_payload(payload, relationships)
```

Comment:
In the name of a single function not doing multiple things I suggest using only `merge_relationships(relationships)` and returning the new, merged relationship which can then be added here to the payload.

Reply:
Agree, changed